File: historyscan-back/src/main/java/io/sch/historyscan/domain/contexts/codebase/clone/Clone.java
package io.sch.historyscan.domain.contexts.codebase.clone;

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI;

@HexagonalArchitectureAPI
@FunctionalInterface
public interface Clone {
    ClonedCodeBase from(CodeBaseToClone codeBaseToClone) throws CodeBaseAlreadyExistsException;
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/clone/Clone.kt
package io.sch.historyscan.domain.contexts.codebase.clone

import io.sch.historyscan.domain.hexagonalarchitecture.HexagonalArchitectureAPI

@HexagonalArchitectureAPI
fun interface Clone {
    @Throws(CodeBaseAlreadyExistsException::class)
    fun from(codeBaseToClone: CodeBaseToClone): ClonedCodeBase
}
